twitter/modules/listener/EventDispatcher.bal

```diff
@@ -16,45 +16,32 @@
 
 import ballerina/log;
 
-class EventDispatcher {
-    private SupportedRemoteFunctionImpl supportedRemoteFunctionImpl = {};
-
-    private SimpleHttpService httpService;
-
-    isolated function init(SimpleHttpService httpService) {
-        self.httpService = httpService;
-
-        string[] methodNames = getServiceMethodNames(httpService);
-
-        foreach var methodName in methodNames {
-            match methodName {
-                "onTweet" => {
-                    self.supportedRemoteFunctionImpl.isOnTweet = true;
-                }
-                "onReply" => {
-                    self.supportedRemoteFunctionImpl.isOnReply = true;
-                }
-                "onReTweet" => {
-                    self.supportedRemoteFunctionImpl.isOnReTweet = true;
-                }
-                "onQuoteTweet" => {
-                    self.supportedRemoteFunctionImpl.isOnQuoteTweet = true;
-                }
-                "onFollower" => {
-                    self.supportedRemoteFunctionImpl.isOnFollower = true;
-                }
-                "onFavourite" => {
-                    self.supportedRemoteFunctionImpl.isOnFavourite = true;
-                }
-                "onDelete" => {
-                    self.supportedRemoteFunctionImpl.isOnDelete = true;
-                }
-                "onMention" => {
-                    self.supportedRemoteFunctionImpl.isOnMention = true;
-                }
-                _ => {
-                    log:printError("Unrecognized method [" + methodName + "] found in the implementation");
-                }
+isolated class EventDispatcher {
+    private final boolean isOnTweet;
+    private final boolean isOnReply;
+    private final boolean isOnReTweet;
+    private final boolean isOnQuoteTweet;
+    private final boolean isOnFollower;
+    private final boolean isOnFavourite;
+    private final boolean isOnDelete;
+    private final boolean isOnMention;
+    private final HttpToTwitterAdaptor adaptor;
+    isolated function init(HttpToTwitterAdaptor adaptor) {
+        self.adaptor = adaptor;
+        
+        string[] methodNames = adaptor.getServiceMethodNames();
+        self.isOnTweet = isMethodAvailable("onTweet", methodNames);
+        self.isOnReply = isMethodAvailable("onTweet", methodNames);
+        self.isOnReTweet = isMethodAvailable("onTweet", methodNames);
+        self.isOnQuoteTweet = isMethodAvailable("onTweet", methodNames);
+        self.isOnFollower = isMethodAvailable("onTweet", methodNames);
+        self.isOnFavourite = isMethodAvailable("onTweet", methodNames);
+        self.isOnDelete = isMethodAvailable("onTweet", methodNames);
+        self.isOnMention = isMethodAvailable("onTweet", methodNames);
```

Comment:
"onTweet" is repeated in all the places.

Reply:
Fixed